domain/access/service/permission.go

```diff
@@ -31,7 +32,13 @@ func NewPermissionService(st PermissionState) *PermissionService {
 
 // CreatePermission gives the user access per the provided spec. All errors
 // are passed through from the spec validation and state layer.
-func (s *PermissionService) CreatePermission(ctx context.Context, spec corepermission.UserAccessSpec) (corepermission.UserAccess, error) {
+func (s *PermissionService) CreatePermission(ctx context.Context, spec corepermission.UserAccessSpec) (_ corepermission.UserAccess, err error) {
```

Comment:
Are the named returns `(_ corepermission.UserAccess, err error)` necessary? Do the signatures have to respect some sort of format?

Reply:
Yes. Otherwise RecordError wouldn't work.